packages/liveblocks-core/src/crdts/LiveObject.ts

```diff
@@ -544,6 +560,32 @@ export class LiveObject<O extends LsonObject> extends AbstractCrdt {
    */
   update(patch: Partial<O>): void {
     this._pool?.assertStorageIsWritable();
+
+    // If detectLargeObjects is enabled, perform a runtime size check now so we
+    // can immediately throw as soon as the max object size is exceeded.
+    if (LiveObject.detectLargeObjects) {
+      const data: Record<string, Json> = {};
+      for (const [key, value] of this.#map) {
+        if (!isLiveNode(value)) {
+          data[key] = value;
+        }
+      }
+      for (const key of Object.keys(patch)) {
+        const value = patch[key];
+        if (value === undefined) continue;
+        if (!isLiveNode(value)) {
+          data[key] = value;
+        }
+      }
+
+      const newSize = new TextEncoder().encode(JSON.stringify(data)).length;
+      if (newSize > MAX_LIVE_OBJECT_SIZE) {
+        throw new Error(
+          `LiveObject size exceeded limit: ${newSize} bytes > ${MAX_LIVE_OBJECT_SIZE} bytes. See https://liveblocks.io/docs/platform/limits#Liveblocks-Storage-limits`
+        );
+      }
```

Comment:
For performance reasons, could we do a quick approximate check by multiplying the string's length by 4 (which returns an upper bound assuming that every character might take up to 4 bytes - as in worst-case UTF-8 encoding) and comparing it to the maximum allowed size? Then, we could perform a more precise check (the one you have currently).

Reply:
Ah yes, I think that's a great idea!